src/main/java/core/org/akaza/openclinica/service/PermissionServiceImpl.java

```diff
@@ -200,6 +200,20 @@ public boolean hasFormAccess(EventCrf ec, Integer formLayoutId, Integer studyEve
         return true;
     }
 
+    public boolean hasFormAccess(EventDefinitionCrf edc,List<String> permissionTagsList) {
```

Comment:
looks like this code is already there in the existing hasFormAccess method. Can you refactor the code such that both methods are sharing this code instead of repeating the logic?

Reply:
done -- both methods are sharing same common code